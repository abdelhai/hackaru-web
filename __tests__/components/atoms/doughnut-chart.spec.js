import Factory from '@/__tests__/__setups__/factory';
import DoughnutChart from '@/components/atoms/doughnut-chart';

describe('DoughnutChart', () => {
  let factory;

  beforeEach(() => {
    factory = new Factory(DoughnutChart, {
      propsData: {
        chartData: {
          labels: ['Review', 'Development'],
          datasets: [{ data: [100, 50], backgroundColor: ['#ff0', '#f00'] }]
        }
      }
    });
  });

  it('render correctly', () => {
    const wrapper = factory.shallow();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('show tooltip text correctly', () => {
    const wrapper = factory.shallow();
    expect(
      wrapper.vm.options.tooltips.callbacks.label(
        { index: 0 },
        { datasets: [{ data: [100] }] }
      )
    ).toBe('01:40');
  });
});
