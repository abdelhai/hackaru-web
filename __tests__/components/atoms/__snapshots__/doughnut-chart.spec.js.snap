// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoughnutChart render correctly 1`] = `
<div
  class=""
>
  <div
    class="chartjs-render-monitor"
    height="400"
    id="doughnut-chart"
    style="display: block;"
    width="400"
  />
</div>
`;
